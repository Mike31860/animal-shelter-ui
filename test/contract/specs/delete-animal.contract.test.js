import { provider } from "./init-pact";
import { AnimalController } from "../../../controllers";
import { Matchers } from "@pact-foundation/pact";

const name = "Copito";

describe('Given An Animal service', () => {
    describe('When a request to delete an specified animal is made', () => {
        beforeAll(async () => {
            await provider.setup();
            await provider.addInteraction({
                state: 'name of animal',
                uponReceiving: 'The Request to delete the animal',
                withRequest: {
                    method: 'DELETE',
                    path: `/animals/${name}`,
                },
                willRespondWith: {
                    status: 204,
                }
            });
        });

        it("Then it should return the right status", async() =>{

            const response = await AnimalController.delete(name);
            expect(response.data).toMatchSnapshot();

            await provider.verify();
        });

        afterAll(async () => {
            await provider.finalize();
        });
    });
});