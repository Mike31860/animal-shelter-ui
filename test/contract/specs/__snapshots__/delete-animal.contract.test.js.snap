// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given An Animal service When a request to delete an specified animal is made Then it should return the right status 1`] = `""`;
